src/common/mfu_io.c

```diff
@@ -57,7 +252,27 @@ int mfu_lchown(const char* path, uid_t owner, gid_t group)
     return rc;
 }
 
-/* calls chmod, and retries a few times if we get EIO or EINTR */
+int daos_chmod(const char *path, mode_t mode, mfu_file_t* mfu_file)
+{
+#ifdef DAOS_SUPPORT
+    int rc;
+    dfs_obj_t *parent = NULL;
+    char *name = NULL, *dir_name = NULL;
+    parse_filename(path, &name, &dir_name);
+    assert(dir_name);
+    rc = dfs_lookup(mfu_file->dfs, dir_name, O_RDWR, &parent, NULL, NULL);
+    if (parent == NULL) {
+        fprintf(stderr, "dfs_lookup %s failed \n", dir_name);
+	return ENOENT;
```

Comment:
It's a line like this one we want to look for.  We'd want to change this to something like:
```suggestion
	errno = ENOENT;
	return -1;
```

Reply:
@adammoody I see, looks like the issue then is just with the wrappers around the DFS calls, so I can update these.

Reply:
Maybe we don't need to set errno here, if we can rely on it being set from the dfs_lookup call.  Though we want to be sure to return -1 from ```mfu_file_chmod``` on any error, since code calling chmod might be doing something like:
```
int rc = mfu_file_chmod(...);
if (rc != -1)
  // success
else
  // error
```

Reply:
@adammoody ok, I added checks in those mfu_file_* functions, if rc was not zero, I set the rc to return -1.

Reply:
@adammoody I am going to update the changes we talked about yesterday in dcp: removing the daos short options, the check for the same pool, etc. Then I will add a check for the num of source paths.

Reply:
> 
> 
> @adammoody ok, I added checks in those mfu_file_* functions, if rc was not zero, I set the rc to return -1.

I don't think all mfu_file_* functions return -1 on error.  We need to check the man page for each call to read what it should return on success and what it should return on failure.

Reply:
@adammoody yeah, well all of the ones I updated say that in the man page (access, chmod, stat, Monod, close, truncate, and closer). The one that differs is lseek, which returns an offset on success, and -1 otherwise.